util/execdetails/execdetails.go

```diff
@@ -366,73 +366,73 @@ func (rrs *ReaderRuntimeStats) String() string {
 	return fmt.Sprintf("rpc num: %v, rpc max:%v, min:%v, avg:%v, p80:%v, p95:%v, proc keys max:%v, p95:%v", size, vMax, vMin, vAvg, vP80, vP95, keyMax, keyP95)
 }
 
-// RuntimeStatsColl collects executors's execution info.
-type RuntimeStatsColl struct {
-	mu          sync.Mutex
-	rootStats   map[string]*RuntimeStats
-	copStats    map[string]*CopRuntimeStats
-	readerStats map[string]*ReaderRuntimeStats
+// RuntimeStats is used to express the executor runtime information.
+type RuntimeStats interface {
+	GetActRows() int64
+	SetRowNum(int64)
+	String() string
 }
```

Comment:
I prefer to remove the method `SetRowNum` and make this interface read-only.

Reply:
yes, I  agree too. But currently somewhere will need to call `SetRowNum`, and there only has a `planID`, I'm not sure just use `RegisterStats` instead of `SetRowNum` is safe in those place.